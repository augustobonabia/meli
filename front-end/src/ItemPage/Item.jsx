import React from 'react';
import PropTypes from 'prop-types';
import ItemPrice, { pricePropType } from '../shared-componets/ItemPrice';
import './index.scss';

function Item(props) {
  const { item } = props;
  let status = '';
  if (item.condition === 'new') {
    status = 'Nuevo';
  } else if (item.condition === 'used') {
    status = 'Usado';
  }

  if (status) {
    status += ' - ';
  }
  status += `${item.sold_quantity} vendidos`;

  return (
    <div className="page-section-container item-details-container">
      <div className="grid">
        <article name={item.id} alt={item.title}>
          <img alt="Imagen del producto" src={item.picture} />
          <section className="product-description">
            <h1>Descripci√≥n del producto</h1>
            <p>{item.description}</p>
          </section>
        </article>
        <aside name="buy info">
          <section className="status">{status}</section>
          <h1>{item.title}</h1>
          <ItemPrice price={item.price} />
          <button type="button" className="buy">Comprar</button>
        </aside>
      </div>
    </div>
  );
}

Item.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    price: pricePropType,
    picture: PropTypes.string.isRequired,
    condition: PropTypes.string.isRequired,
    free_shipping: PropTypes.bool.isRequired,
    description: PropTypes.string.isRequired,
    sold_quantity: PropTypes.number.isRequired,
    location: PropTypes.string,
  }).isRequired,
};

export default Item;
