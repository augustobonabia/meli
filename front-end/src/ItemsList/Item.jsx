import React from 'react';
import PropTypes from 'prop-types';
import ItemPrice, { pricePropType } from '../shared-componets/ItemPrice';

function Item(props) {
  const { item } = props;

  const renderFreeShipping = (hasFreeShipping) => {
    if (hasFreeShipping) {
      return <img className="free-shipping" alt="¡Envío gratis!" src="../shipping.png" />;
    }

    return null;
  };

  return (
    <li className="item-container">
      <article alt="Producto" className="item">
        <img className="thumbnail" alt="Pequeña imagen del producto" src={item.picture} />
        <div className="item-content">
          <section alt="Información principal" className="abstract">
            <div className="price-and-shipping">
              <ItemPrice price={item.price} />
              {renderFreeShipping(item.free_shipping)}
            </div>
            <h1 alt="Título">{item.title}</h1>
          </section>
          <section alt="Información adicional" className="additional-info">
            <span alt="Ubicación del vendedor">{item.location}</span>
          </section>
        </div>
      </article>
    </li>
  );
}

Item.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    price: pricePropType.isRequired,
    picture: PropTypes.string.isRequired,
    condition: PropTypes.string.isRequired,
    free_shipping: PropTypes.bool.isRequired,
    location: PropTypes.string,
  }).isRequired,
};

export default Item;
