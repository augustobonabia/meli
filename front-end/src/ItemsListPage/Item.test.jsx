import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Item from './Item';

it('Un item del listado de bÃºsqueda se renderiza correctamente', () => {
  const item = {
    id: 'MLA123',
    title: 'Mesa de ping pong',
    price: {
      amount: 34999,
      decimals: 99,
      currency: '$',
    },
    picture: 'una-imagen.png',
    condition: 'new',
    free_shipping: true,
    location: 'Capital Federal',
  };

  const tree = renderer
    .create(<Router><Item item={item} /></Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
