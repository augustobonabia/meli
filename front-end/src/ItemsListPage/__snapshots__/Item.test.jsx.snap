// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Un item del listado de búsqueda se renderiza correctamente 1`] = `
<li
  className="item-container"
>
  <a
    href="#/items/MLA123"
    onClick={[Function]}
  >
    <article
      alt="Producto"
      className="item"
    >
      <img
        alt="Pequeña imagen del producto"
        className="thumbnail"
        src="una-imagen.png"
      />
      <div
        className="item-content"
      >
        <section
          alt="Información principal"
          className="abstract"
        >
          <div
            className="price-and-shipping"
          >
            <span
              alt="Precio"
              className="price"
            >
              <span
                alt="Moneda"
                className="currency"
              >
                $
              </span>
              <span
                alt="Monto"
                className="amount"
              >
                34.999
              </span>
              <span
                className="cents"
              >
                99
              </span>
            </span>
            <img
              alt="¡Envío gratis!"
              className="free-shipping"
              src="./shipping.png"
            />
          </div>
          <h1
            alt="Título"
          >
            Mesa de ping pong
          </h1>
        </section>
        <section
          alt="Información adicional"
          className="additional-info"
        >
          <span
            alt="Ubicación del vendedor"
          >
            Capital Federal
          </span>
        </section>
      </div>
    </article>
  </a>
</li>
`;
